src/Lecture1.hs

```diff
@@ -94,7 +101,10 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str = take (end - start' + 1) (drop start' str)
```

Comment:
I see that with the following input, for example, the output could be wrong though:

```haskell
> subString (-5) 0 "abc"
"a"
```

(I think we need to add this example to tests as well 🙂 )